src/test/java/com/adobe/target/edge/client/ondevice/collator/CustomParamsCollatorTest.java

```diff
@@ -43,4 +43,37 @@ public void testCollator() {
     assertEquals("BUZ", result.get("BAZ"));
     assertEquals("buz", result.get("BAZ" + CustomParamsCollator.LOWER_CASE_POSTFIX));
   }
+
+  @Test
+  public void testCustomDotNotation () {
+    VisitorProvider.init("testOrgId");
+    Map<String, String> params =
+      new HashMap<String, String>() {
+        {
+          put("foo", "bar");
+          put("BAZ", "BUZ");
+          put("dot.notation", "isConfusing");
+          put("first.second.third", "value");
+          put("first.second.wonky", "DONKEY");
+          put("this..should..be", "ignored");
```

Comment:
what would happen with these properties...

put(".something", "aaa");
put("=cranky .chicken.", "bbb");

Reply:
As it is currently written these tests would cause an error. Need to change the code so that these keys without proper dot notation (trailing or starting dots) will not be turned into nested objects